src/overlays/actors/ovl_En_Po_Relay/z_en_po_relay.c

```diff
@@ -21,35 +44,367 @@ const ActorInit En_Po_Relay_InitVars = {
+            vec.z = (Math_Coss(func_8005A9F4(ACTIVE_CAM) + 0x4800) * 23.0f) + this->actor.posRot.pos.z;
+        }
+        EffectSsDeadDb_Spawn(globalCtx, &vec, &D_80AD8D30, &D_80AD8D3C, this->actionTimer * 10 + 80, 0, 255, 255, 255,
+                             255, 0, 0, 255, 1, 9, 1);
+        vec.x = (this->actor.posRot.pos.x + this->actor.posRot.pos.x) - vec.x;
+        vec.z = (this->actor.posRot.pos.z + this->actor.posRot.pos.z) - vec.z;
+        EffectSsDeadDb_Spawn(globalCtx, &vec, &D_80AD8D30, &D_80AD8D3C, this->actionTimer * 10 + 80, 0, 255, 255, 255,
+                             255, 0, 0, 255, 1, 9, 1);
+        vec.x = this->actor.posRot.pos.x;
+        vec.z = this->actor.posRot.pos.z;
+        EffectSsDeadDb_Spawn(globalCtx, &vec, &D_80AD8D30, &D_80AD8D3C, this->actionTimer * 10 + 80, 0, 255, 255, 255,
+                             255, 0, 0, 255, 1, 9, 1);
+        if (this->actionTimer == 1) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_EXTINCT);
+        }
+    }
+    if (Math_ApproxF(&this->actor.scale.x, 0.0f, 0.001f) != 0) {
+        if (this->hookshotSlotFull != 0) {
+            sp60.x = this->actor.posRot.pos.x;
+            sp60.y = this->actor.groundY;
+            sp60.z = this->actor.posRot.pos.z;
+            if (gSaveContext.timer1Value < gSaveContext.dampeRaceTime) {
+                gSaveContext.dampeRaceTime = gSaveContext.timer1Value;
+            }
+            if (Flags_GetCollectible(globalCtx, this->actor.params) == 0 && gSaveContext.timer1Value <= 60) {
+                Item_DropCollectible2(globalCtx, &sp60, (this->actor.params << 8) + (0x4000 | ITEM00_HEART_PIECE));
+            } else {
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00, sp60.x, sp60.y, sp60.z, 0, 0, 0, 2);
+            }
+        } else {
+            Flags_SetTempClear(globalCtx, 4);
+            gSaveContext.dampeRaceTime = gSaveContext.timer1Value;
+        }
+        Actor_Kill(&this->actor);
+    }
+    this->actor.scale.y = this->actor.scale.x;
+    this->actor.scale.z = this->actor.scale.x;
+    this->actor.posRot.pos.y += 10.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Po_Relay/func_80AD7C64.s")
+void EnPoRelay_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnPoRelay* this = THIS;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Po_Relay/func_80AD8174.s")
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    this->actionFunc(this, globalCtx);
+    Actor_MoveForward(&this->actor);
+    EnPoRelay_CorrectY(this);
+    func_8002E4B4(globalCtx, &this->actor, 0.0f, 27.0f, 60.0f, 4);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_SetHeight(&this->actor, 50.0f);
+    if (this->unk_195 != 0) {
+        this->unk_195 -= 1;
+    }
+    if (this->unk_195 == 0) {
+        this->unk_195 = 32;
+    }
+    this->eyeImageIdx++;
```

Comment:
id prob go with texture index as opposed to image

Reply:
I named it this to mirror the variable of the same name in En_Tk, I can change both if you think it's definitely better?

Reply:
yea that name is from a while back. it indexes an array of segmented addresses for textures so i think itd be best